src/abstractions/Helpers/EnumHelpers.cs

```diff
@@ -154,5 +155,29 @@ private static bool TryGetFieldValueName(Type type, string rawValue, out string
             }
             return false;
         }
+
+        /// <summary>
+        /// Gets the enum string representation of the given value. Looks up if there is an <see cref="EnumMemberAttribute"/> and returns the value if found, otherwise returns the enum name in camel case.
+        /// </summary>
+        /// <typeparam name="T">The Enum type</typeparam>
+        /// <param name="value">The enum value</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentException">If value is null</exception>
+#if NET5_0_OR_GREATER
+        public static string? GetEnumStringValue<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)] T>(T value) where T : struct, Enum
+#else
+        public static string? GetEnumStringValue<T>(T value) where T : struct, Enum
+#endif
+        {
+            var type = typeof(T);
+
+            if(Enum.GetName(type, value) is not { } name)
+                throw new ArgumentException($"Invalid Enum value {value} for enum of type {type}");
+
+            if(type.GetField(name)?.GetCustomAttribute<EnumMemberAttribute>() is { } attribute)
+                return attribute.Value;
+
+            return name.ToFirstCharacterLowerCase();
```

Comment:
Oversight?

Reply:
Thanks for highlighting this. I am hesitant to drop this one as this only happens when there is no `EnumMember` attribute where it tries to camelCase the property. 
Given that the convention is to have the enum members to start with uppercase in C#, this means that member will typically be serialized in pascal case which is not an expectation we've set in previous releases.

Reply:
this is a thing only dotnet is doing today.
Assuming we don't have special casing in the generation for dotnet to add/remove the enum member attributes (i.e. it follows the same logic as the other languages that are upper camel cased) we should be fine.
But yes, we should make sure the generation is aligned before we make that change.